core/api/service/system/requests/name.cpp

```diff
@@ -19,11 +19,10 @@ namespace kagome::api::system::request {
   }
 
   outcome::result<std::string> Name::execute() {
-    // TODO(xDimon): Ensure if implementation is correct, and remove exception
-    throw jsonrpc::InternalErrorFault(
-        "Internal error: method is known, but not yet implemented");
-
-    return api_->getConfig()->name();
+    // In Polkadot it is just hardcoded string of client's implementation name:
+    // https://github.com/paritytech/polkadot/blob/c68aee352b84321b6a5691d38e20550577d60a45/cli/src/command.rs#L34
+    // So I assume we can safely use our client's name ehre
+    return "Soramitsu Kagome";
```

Comment:
I'm suggesting to collect same hardcoded constants and literals in special .h file. Perhaps, in cmake generated.

Reply:
Do we need exactly ```std::string``` result? Maybe we can use ```std::string_view```. It seems redundant to build a string with a literal constant data not to be modified later.
```c++
return "Soramitsu Kagome";

Reply:
> I'm suggesting to collect same hardcoded constants and literals in special .h file. Perhaps, in cmake generated
Sounds like out of scope for that PR

Reply:
> Do we need exactly std::string result? Maybe we can use std::string_view. It seems redundant to build a string with a literal constant data not to be modified later.

Unfortunatelly, yes. Result should be eventually converted into jsonrpc::Value which can only store string. So it is easier to return string